### PR TITLE
fix: [SRM-12531]: Add edge changes for AWS Prometheus

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

1. Make selection of data source type mandate
2. Disable connector until data source type is selected
3. Reset connector once data source type changes

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vasuki-kalluri — DISMISSED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 1]
```diff
@@ -54,6 +59,62 @@ export const validateDuplicateIdentifier = (values: DefineHealthSourceFormInterf
...
+}
+
+export const getIsConnectorDisabled = ({
+  isEdit,
+  connectorRef,
+  sourceType,
+  isDataSourceTypeSelectorEnabled,
+  dataSourceType
+}: ConnectorDisableFunctionProps): boolean => {
+  if (isEdit && connectorRef) {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:109]: all if and else if return true , is this expected ?

Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:109]: Yes, just to keep the code clean and readable.

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 2]
```diff
@@ -25,7 +25,12 @@ import {
...
 } from './DefineHealthSource.constant'
-import type { DefineHealthSourceFormInterface } from './DefineHealthSource.types'
+import type {
+  ConnectorDisableFunctionProps,
+  DataSourceTypeValidateFunctionProps,
+  DefineHealthSourceFormInterface,
+  FormValidationFunctionProps
+} from './DefineHealthSource.types'
 
 export const validate = (getString: UseStringsReturn['getString']) => {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:35]: please define the return type for all functions.

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 3]
```diff
@@ -54,6 +59,62 @@ export const validateDuplicateIdentifier = (values: DefineHealthSourceFormInterf
   if (healthSourceList?.some(item => item.identifier === healthSourceIdentifier)) {
     return { healthSourceName: 'identifier already exist' }
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:60]: use strings here.

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts [hunk 4]
```diff
@@ -54,6 +59,62 @@ export const validateDuplicateIdentifier = (values: DefineHealthSourceFormInterf
...
+}: FormValidationFunctionProps): Record<string, string> => {
+  let errors = {}
+
+  if (!isEdit) {
+    errors = validateDuplicateIdentifier(values)
+  }
+
+  if (
+    isDataSourceTypeNotValid({
+      dataSourceType: values?.dataSourceType,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts:88]: destructure values.

Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts]: Done

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts (hunk 3 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts]: Done

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.utils.ts]: Done

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx [hunk 5]
```diff
@@ -159,7 +160,13 @@ function DefineHealthSource(props: DefineHealthSourceProps): JSX.Element {
           placeholder={getString('cv.healthSource.connectors.selectConnector', {
             sourceType: getConnectorPlaceholderText(formik?.values?.sourceType, formik?.values?.dataSourceType)
           })}
-          disabled={isEdit ? !!formik?.values?.connectorRef && isEdit : !formik?.values?.sourceType}
+          disabled={getIsConnectorDisabled({
+            isEdit,
+            connectorRef: formik?.values?.connectorRef,
```
[the remark this answers is not in the document]
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx]: Done

**Review by harshilbits — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**